shutdown/legacy.go

```diff
@@ -0,0 +1,84 @@
+package shutdown
+
+import (
+	"github.com/ThomasObenaus/go-base/shutdown/health"
+	"github.com/ThomasObenaus/go-base/shutdown/log"
+	"github.com/ThomasObenaus/go-base/shutdown/signal"
+	"github.com/ThomasObenaus/go-base/shutdown/stop"
+	"github.com/rs/zerolog"
+)
+
+type ShutdownHandler struct {
+	stoppableItems stopIF
+	signalHandler  signalHandlerIF
+	log            logIF
+	health         healthIF
+}
+
+func InstallHandler(orderedStopables []stop.Stoppable, logger zerolog.Logger) *ShutdownHandler {
+	shutdownHandler := &ShutdownHandler{
+		stoppableItems: &stop.OrderedStoppableList{},
+		log:            log.ShutdownLog{Logger: logger},
+		health:         &health.Health{},
+	}
+
+	for _, stopable := range orderedStopables {
+		err := shutdownHandler.stoppableItems.AddToBack(stopable)
+		if err != nil {
+			logger.Error().Err(err).Msgf("unexpected error adding stoppable to internal list")
+			return nil
+		}
+	}
+
+	handler := signal.NewDefaultSignalHandler(shutdownHandler)
+	shutdownHandler.signalHandler = handler
+
+	return shutdownHandler
+}
+
+func (h *ShutdownHandler) Register(stoppable stop.Stoppable, front ...bool) {
+	addToBack := isFirstBoolUndefinedOrFalse(front)
+
+	if addToBack {
+		err := h.stoppableItems.AddToBack(stoppable)
+		if err != nil {
+			serviceName := stoppable.String()
+			h.log.LogCanNotAddService(serviceName)
+		}
+		return
```

Comment:
this should return an error

Reply:
This would break the current API. Before it would silently not stop Stoppables which where added while a shutdown is in progress. Now it at least logs something.

But I can change the API and consumers of this would have to handle the error then.